packages/meditrak-server/src/externalApiSync/ExternalApiSyncQueue.js

```diff
@@ -78,8 +78,16 @@ export class ExternalApiSyncQueue {
 
   processUpdates = async changes => {
     const validUpdates = await this.validator.getValidUpdates(changes);
-    const changeDetails = await this.detailGenerator.generateDetails(validUpdates);
-    return this.persistToSyncQueue(validUpdates, changeDetails);
+    try {
+      const changeDetails = await this.detailGenerator.generateDetails(validUpdates);
+      return this.persistToSyncQueue(validUpdates, changeDetails);
+    } catch (e) {
+      // Something went wrong with generating change details, possibly because the entity hierarchy
+      // cache is still being built as the result of an entity change. Put this batch of changes
+      // back on the queue to attempt processing again
+      this.unprocessedChanges.push(...validUpdates);
```

Comment:
The error handling here makes sense in the general case! Let's also keep in mind the cases where this could continuously fail for a legit reason. We may want to handle them in a different way (not retry).

One approach could be to retry a max number of times per batch of updates. This would also require assigning batches an id of sort.

We can probably ignore this for now and deal with it if such a situation arises

Reply:
Yep it's a good point. We should notice if this happens for real - it would be the same as the current behaviour, which is that the sync queue would get blocked completely. This doesn't make things any worse ;-)

Reply:
Good point! An additional justification for not doing anything extra at the moment 🙌